hostapd/init.sls

```diff
@@ -23,7 +23,12 @@ hostapd_activate:
   file.replace:
     - name: {{ map.defaults_file }}
     - pattern: "^(|#)DAEMON_CONF=.*$"
-    - repl: "DAEMON_CONF='{% for card in salt['pillar.get']('hostapd:cardlist', {}).keys() %}{{ card2conf(card, map) }} {% endfor %}'"
+    - repl: "DAEMON_CONF='
+        {%- for card in salt['pillar.get']('hostapd:cardlist', {}).keys() -%}
+          {{ card2conf(card, map) }}
+          {%- if not loop.last %} {% endif -%}
+        {%- endfor -%}
+'"
```

Comment:
I understand the issue with trailing space, but also think this looks over-complicated. Better to separate Jinja code and apply filter to a string. For example:

```jinja
{%- set d_conf = [] %}
{%- for card in salt['pillar.get']('hostapd:cardlist', {}) %}
  {%- set entry %}{{ card2conf(card, map) }}{% endset %}
  {%- do d_conf.append(entry) %}
{%- endfor %}
```

```yaml
hostapd_activate:
  file.replace:
    - name: {{ map.defaults_file }}
    - pattern: "^(|#)DAEMON_CONF=.*$"
    - repl: DAEMON_CONF='{{ d_conf|join(" ") }}'
```

As you can see, the Jijna code is not much verbose, but the `repl` parameter argument looks the way better and there is no mess with whitespace control.

Reply:
@vutny I fully agree. :-) PR updated.

Reply:
how backwards compatible is this? My feeling is that jinja filters is something from newer salt versions.

Reply:
@aboe76 Yes, you're absolutely right! Just the problem lies within block assignments, which are only available since Jinja 2.8. Not all platforms have this or later version.

@alxwr
I suggest to throw away the `card2conf` macro and simply calculate the map upfront. This will save us from fetching pillar data twice, because it is heavy operation. Like this:
```jinja
{#- Create card to config mapping #}
{%- set daemon_conf = {} %}
{%- for card in salt['pillar.get']('hostapd:cardlist', {}) %}
{%-   set conf = '%s/%s_%s.conf'|format(map.conf_dir, map.conf_file|replace('.conf', ''), card) %}
{%-   do daemon_conf.update({card: conf}) %}
{%- endfor %}
```
Then replacement value becomes as simple as:
```jinja
DAEMON_CONF='{{ daemon_conf.values()|join(" ") }}'
```
Finally, later in the code ([line 42](https://github.com/saltstack-formulas/hostapd-formula/pull/5/files#diff-956796f2376ba3cea87f99e0ff32c19fR42)) we would iterate just over an existing cards:
```yaml
{%- for card, conf in daemon_conf|dictsort %}

hostapd_config_{{ card }}:
  file.managed:
    - name: {{ conf }}
    ...

{%- endfor %}
```

Reply:
@vutny Great suggestion, thanks! I went a bit further and removed the call to `pillar.get` from the template too.